### DISCUSSION TITLE
OpenGlContextBase - glGenerateMipmap to add textures on vertices

### DISCUSSION THREAD

**Discussion by markusalbrecht-procam** [AvaloniaUI/Avalonia#14980] [General]
Hi,
we have just added the import of OBJ data into our AvaloniaUI 3D App prototype. 
The problem we encounter now is we cannot find any way to use the function for glGenerateMipmap as described in:
https://registry.khronos.org/OpenGL-Refpages/es3.1/html/glGenerateMipmap.xhtml

OBJ files can have also textures which are displayed on the triangle surface via MTL fles.

Does anyone have experience with adding textures in 3D in OpenES inside AvaloniaUI ?

We only found OpenTK samples but there is nothing in the web related to OpenES :(.
https://stackoverflow.com/questions/34006850/c-opentk-problems-applying-shaders-to-loaded-texture-texture-is-scaled-down

BR
Markus

**Comment by kekekeks** [comment]
To call extra functions you can use [Silk.NET](https://github.com/kekekeks/Avalonia-Silk.NET-Example/tree/master) or other OpenGL API binding.

**Reply by markusalbrecht-procam** [reply]
THANKS. It can be used in combination with the OpenGlRenderBase of AvaloniaUi I guess ?
